server/polar/subscription/endpoints.py

```diff
@@ -129,3 +138,73 @@ async def create_csv() -> AsyncGenerator[str, None]:
         media_type="text/csv",
         headers={"Content-Disposition": f"attachment; filename={filename}"},
     )
+
+
+@router.patch(
+    "/{id}",
+    summary="Update Subscription",
+    response_model=SubscriptionSchema,
+    responses={
+        200: {"description": "Subscription updated."},
+        403: {
+            "description": (
+                "Subscription is already canceled "
+                "or will be at the end of the period."
+            ),
+            "model": AlreadyCanceledSubscription.schema(),
+        },
+        404: SubscriptionNotFound,
+    },
+)
+async def update(
+    id: SubscriptionID,
+    subscription_update: SubscriptionUpdate,
+    auth_subject: auth.SubscriptionsWrite,
+    session: AsyncSession = Depends(get_db_session),
+) -> Subscription:
+    """Update a subscription of the authenticated customer or user."""
+    subscription = await subscription_service.user_get(session, auth_subject, id)
+    if subscription is None:
+        raise ResourceNotFound()
+
+    log.info(
+        "subscription.update",
+        id=id,
+        customer_id=auth_subject.subject.id,
+        updates=subscription_update,
+    )
+    return await subscription_service.update(
+        session, subscription, updates=subscription_update
+    )
+
+
+@router.delete(
+    "/{id}",
+    summary="Revoke Subscription",
+    response_model=SubscriptionSchema,
+    responses={
+        200: {"description": "Subscription revoked."},
+        403: {
+            "description": "This subscription is already revoked.",
+            "model": AlreadyCanceledSubscription.schema(),
+        },
+        404: SubscriptionNotFound,
+    },
+)
+async def revoke(
```

Comment:
Oh nice! I like the fact that `DELETE` is for cancelling immediately vs. `PATCH` is used to plan a cancellation 👍